### ISSUE TITLE
Meshcentral functions (take control / file manager) very slow to load in Firefox

### ISSUE THREAD

**Issue by af7567** [wh1te909/tacticalrmm#840]
**Server Info (please complete the following information):**
 - OS: Ubuntu 20.04.3
 - Browser: Firefox 94.0.2
 - RMM Version (as shown in top left of web UI):v0.10.3

**Installation Method:**
  - [*] Standard

**Describe the bug**
After installation / upgrade of TRRM everything works fine, but after a day or so the "Take Control" option takes a long time (1 minute +) to display the Connect button in the Meshcentral frame - sometimes it never displays it and I just have a black area under the "Agent Status" top bar. If I switch to Chromium then it loads straight away.

I am not sure if this is a Firefox bug or an nginx configuration problem. I disabled all Firefox extensions and Tracking Protection but it still happens.

Installation was done on a fresh Ubuntu 20.04 install using the install.sh script and all updates have been installed with apt.

If I visit https://mesh.example.com/ in Firefox then "My Devices" will stay empty for 15s to 2minutes. But if I directly access the IP of meshcentral on port 4430 it will load the devices instantly every time. So it looks like Firefox and nginx aren't working well together.

The Firefox Network Monitor shows the long pause happening just before a HTTP status of 101 Switching Protocols when requesting wss://mesh.example.com/control.ashx. So it seems to be something to do with WebSockets and nginx.

**Comment by silversword411** [comment]
This would be a meshcentral issue and probably not something we can fix. 

Open this in the meshcentral issues. They fix there, we'll get the fix here :)

**Comment by silversword411** [comment]
We can leave this open for a bit, let us know if they see the same issue over there.

**Comment by joeldeteves** [comment]
Check your Firefox proxy settings:

https://support.mozilla.org/en-US/kb/connection-settings-firefox

Ensure it is set to **No Proxy:**

![2020-05-20-23-15-05-911af7](https://user-images.githubusercontent.com/33190570/148996456-36b04184-a739-4e67-a9d6-c4282287226c.png)

**Comment by af7567** [comment]
> Check your Firefox proxy settings:

Good idea. I will give it a try.
It was definitely not using a proxy, but was set to "Use system proxy settings".

I have been using Chromium instead of Firefox for a few weeks now but have not been able to reproduce this bug there.

My best guess is that Firefox is waiting for some exisiting connections to close before being able to open another connection to the same IP address/server. But my max connections per server is set to the default of 6 in both Firefox and Chromium.

It could even be a bug in Firefox where an open connection doesn't get closed after the PC has been to sleep and resumed, but that is just another guess :)

**Comment by joeldeteves** [comment]
Yes, 'System proxy setting' will take the proxy setting from the legacy Windows IE control panel. It is probably set to 'auto' there as well - best way to troubleshoot is to make sure it's set to **No Proxy.**

**Comment by silversword411** [comment]
> legacy Windows IE control panel

...and that's a known time-sink of waiting for things to error out before moving on.

**Comment by af7567** [comment]
I have tested Firefox again today with "no proxy" set but already had to wait a couple of minutes for the MeshCentral connect button to appear. So unfortunately that didn't fix it. I am using Firefox in Linux, I will have to test it in Windows Firefox to see how it works there.

Firefox had only been running for about an hour, and the PC hasn't been to sleep so I guess that's not what caused it.
